### PR TITLE
chore: [IAI-206] Add missing styles to `IOBadge`

### PR DESCRIPTION
**Pull request by fabriziofff**
## Short description
This pr adds some missing styles to `IOBadge`, in order to facilitate its use in place of `native-base Badge`.
Schermata 2022-06-03 alle 16 38 21

## List of changes proposed in this pull request
- Add missing colors to `IOBadge`
- Change left & right padding to `8`
- Add border (when required)
- Added the `IOBadge` section in the Showroom

## How to test
👀

### CODE REVIEW

File: ts/components/core/IOBadge.tsx [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
-  labelColor?: "white" | "blue";
+  labelColor?: Extract<IOColorType, "bluegreyDark" | "blue" | "white" | "red">;
 };
 
 const commonBadgeStyles = StyleSheet.create({
   badge: {
-    paddingLeft: 15,
-    paddingRight: 15
+    paddingLeft: 8,
+    paddingRight: 8
```
Comment (emiliopavia) [inline ts/components/core/IOBadge.tsx:16]: @fabriziofff did you change the padding also for non-small badges on purpose?

**Review by emiliopavia — CHANGES_REQUESTED** [review]

File: ts/components/core/IOBadge.tsx (hunk 1 above)
Comment (fabriziofff) [reply ts/components/core/IOBadge.tsx:16]: @emiliopavia yes, I followed the specification provided in https://www.figma.com/file/GBRk8f5WGyaX29lahpT5fM/App-IO---Design-System?node-id=154%3A342. The non-small badges are not provided by design, but I changed the padding in order to have the same (similar) style.

Comment (fabriziofff) [reply ts/components/core/IOBadge.tsx:16]: Maybe @thisisjp could help us to undestand if the same padding is ok?
immagine

Comment (thisisjp) [reply ts/components/core/IOBadge.tsx:16]: Figma is always the source of truth :) We'll review all the components in any case sooner or later...

(btw: where did you see this small badge design? it's not part of our library 👀 )

cc @dmnplb

Comment (fabriziofff) [reply ts/components/core/IOBadge.tsx:16]: I've used https://www.figma.com/file/GBRk8f5WGyaX29lahpT5fM/App-IO---Design-System?node-id=154%3A342 as reference 😄 Did I used the wrong file?

Comment (fabriziofff) [reply ts/components/core/IOBadge.tsx:16]: @emiliopavia we agreed with @thisisjp to keep the same padding. In a next iteration all the "non-small" badges should be removed from the codebase.

**Pull request merged**